### PR TITLE
ddl: add the reason for the unsupported modify column error.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@zimulala @tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -896,43 +896,49 @@ func (d *ddl) DropColumn(ctx context.Context, ti ast.Ident, colName model.CIStr)
...
 		}
 	default:
 		if origin.Tp == to.Tp {
-			return true
+			return nil
 		}
-
-		return false
 	}
+	msg := fmt.Sprintf("column type %s", to.Tp)
```
Comment (zimulala) [inline ddl/ddl_api.go]: Add origin type information.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**